### PR TITLE
nixos/shiori: harden service with systemd

### PR DESCRIPTION
**Pull request by minijackson**
###### Motivation for this change

It's best when it's hardened

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [x] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

**Review by bjornfor — APPROVED** [review]

File: nixos/modules/services/web-apps/shiori.nix [hunk 1]
```diff
@@ -39,9 +39,57 @@ in {
 
       serviceConfig = {
         ExecStart = "${package}/bin/shiori serve --address '${address}' --port '${toString port}'";
+
         DynamicUser = true;
-        Environment = "SHIORI_DIR=/var/lib/shiori";
+        Environment = [ "SHIORI_DIR=/var/lib/shiori" ];
```
Comment (aanderse) [inline nixos/modules/services/web-apps/shiori.nix:44]: I usually prefer to use attribute sets like so:
```
systemd.services.shiori.environment.SHIORI_DIR = "/var/lib/shiori";
```

Of course either way works.

@bjornfor this PR looks good and has your approval, any reason we should hold back on merging?

Comment (minijackson) [reply nixos/modules/services/web-apps/shiori.nix:44]: I don't mind changing it. Out of curiosity though, is there an advantage to doing it this way, apart from allowing other modules to easily access it?

Comment (aanderse) [reply nixos/modules/services/web-apps/shiori.nix]: Not much more of an advantage in this case, aside from what you said. In general using attribute sets over lists allows greater ability to override of course, so I favour that.

Like I said though, either way works.

**Comment by aanderse** [comment]
@minijackson I wonder if you would be interested in submitting a PR upstream to add a `systemd` unit so others could benefit.

@bjornfor I'll leave to you to merge this because I'm not familiar with this software.

**Review by aanderse — APPROVED** [review]

**Comment by bjornfor** [comment]
> @bjornfor I'll leave to you to merge this because I'm not familiar with this software.

I'm not either :-)

**Pull request merged**